aii-opennebula/src/main/resources/vmtemplate.tt

```diff
@@ -69,4 +69,22 @@ DATA = "<vcpu>[% vcpus %]</vcpu><cpu mode='host-passthrough'><topology sockets='
 TYPE = "kvm"
 [%-    END %]
 ]
+[%- FOREACH item IN system.opennebula.pci %]
+PCI = [
+[%    FILTER indent -%]
+[%    INCLUDE 'aii-opennebula/pci_level1.tt' 
+               data=item -%]
+[%-   END -%]
+]
+[%- END %]
+[%- FOR pair IN hardware.cards.ib.pairs %]
```

Comment:
ok perfect, it makes sense and it's quite generic, I will include a loop to search for all pci devices.